src/python/gem5/components/processors/abstract_core.py

```diff
@@ -116,7 +112,38 @@ def connect_interrupt(
     @abstractmethod
     def get_mmu(self) -> BaseMMU:
         """Return the MMU for this core.
-
         This is used in the board to setup system-specific MMU settings.
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def set_simpoint(
```

Comment:
Good job working with my latest changes!

Everything here looks good but I'd add a helpful message to the `NotImplementedError`, something like:

```
raise NotImplementedError("This core type does not support simpoints")
```

I'm thinking about the case where we have cores that are not BaseCPUCores. If someone tries to set and use simpoints a vanilla `NotImplementedError` might be confusing.